src/modules/Game.class.js

```diff
@@ -23,22 +23,156 @@ class Game {
   constructor(initialState) {
     // eslint-disable-next-line no-console
     console.log(initialState);
+
+    // Ініціалізуємо гральну дошку
+    this.defaultState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ];
+
+    this.board = initialState || JSON.parse(JSON.stringify(this.defaultState));
+    this.score = 0; // Початкови рахунок
+    this.status = 'idle'; // Початковий статус гри
+  }
+
+  moveLeft() {
+    this.checkLose();
+
+    for (let i = 0; i < 4; i++) {
+      let newRow = this.board[i].filter((value) => value !== 0);
+
+      for (let j = 0; j < newRow.length - 1; j++) {
+        if (newRow[j] === newRow[j + 1]) {
+          newRow[j] *= 2;
+          this.score += newRow[j];
+          newRow[j + 1] = 0;
+        }
+      }
+
+      newRow = newRow.filter((value) => value !== 0);
+
+      while (newRow.length < 4) {
+        newRow.push(0);
+      }
+
+      this.board[i] = newRow;
+    }
+
+    this.addRandomTile();
+  }
+  moveRight() {
+    this.checkLose();
+
+    for (let i = 0; i < 4; i++) {
+      let newRow = this.board[i]
+        .slice()
+        .reverse()
+        .filter((value) => value !== 0);
+
+      for (let j = 0; j < newRow.length - 1; j++) {
```

Comment:
code in this method is almost the same, but you do reverse before and after manipulations. Try to add reusable code to a separate function and reuse it instead of duplication